py-rattler/rattler/lock/package.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import annotations
+from typing import Optional, Tuple
+from rattler.lock.pypi import PypiPackageData, PypiPackageEnvironmentData
+
+from rattler.rattler import PyLockPackage
+from rattler.repo_data.record import RepoDataRecord
+
+
+class LockPackage:
+    _package: PyLockPackage
+
+    def satisfies(self, spec: str) -> bool:
+        return self._package.satisfies(spec)
```

Comment:
I think this method is ambiguous because the spec is either a Pypi requirement or a Conda matchspec. Maybe it makes more sense to have this method on `RepoDataRecord` and `PyPiPackageData`?